src/pages/TeachersUnite/IntroSchoolPrincipalPage.js

```diff
@@ -0,0 +1,136 @@
+import React, {useCallback} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import Str from 'expensify-common/lib/str';
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import ScreenWrapper from '../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../components/HeaderWithBackButton';
+import Form from '../../components/Form';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import TextInput from '../../components/TextInput';
+import Text from '../../components/Text';
+import styles from '../../styles/styles';
+import * as ErrorUtils from '../../libs/ErrorUtils';
+import ROUTES from '../../ROUTES';
+import Navigation from '../../libs/Navigation/Navigation';
+import TeachersUnite from '../../libs/actions/TeachersUnite';
+import useLocalize from '../../hooks/useLocalize';
+
+const propTypes = {
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    loginList: {},
+};
+
+function IntroSchoolPrincipalPage(props) {
+    const {translate} = useLocalize();
+    /**
+     * @param {Object} values
+     * @param {String} values.firstName
+     * @param {String} values.partneruserid
+     * @param {String} values.lastName
+     */
+    const onSubmit = (values) => {
+        TeachersUnite.addSchoolPrincipal(values.firstName.trim(), values.partneruserid.trim(), values.lastName.trim());
```

Comment:
```suggestion
        TeachersUnite.addSchoolPrincipal(values.firstName.trim(), values.partnerUserID.trim(), values.lastName.trim());
```

Please rename these throughout the code